packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/service/testable/ServiceTestDataState.ts

```diff
@@ -326,6 +331,63 @@ export class ConnectionTestDataState {
     }
   }
 
+  *generateSkeletonTestData(): GeneratorFn<void> {
+    try {
+      this.generatingSkeletonTestDataState.inProgress();
+      const connection = guaranteeNonNullable(
+        this.resolveConnectionValue(this.connectionData.connectionId),
+        `Unable to resolve connection ID '${this.connectionData.connectionId}`,
+      );
+      if (connection instanceof DatabaseConnection) {
+        const serviceExecutionParameters = guaranteeNonNullable(
+          this.testDataState.testSuiteState.testableState.serviceEditorState
+            .executionState.serviceExecutionParameters,
+        );
+        if (serviceExecutionParameters.runtime instanceof RuntimePointer) {
+          const testDataGenerationResult =
+            (yield this.editorStore.graphManagerState.graphManager.generateTestData(
+              getExecutionQueryFromRawLambda(
+                serviceExecutionParameters.query,
+                this.parametersState.parameterStates,
+                this.editorStore.graphManagerState,
+              ),
+              serviceExecutionParameters.mapping.path,
+              serviceExecutionParameters.runtime.packageableRuntime.value.path,
+              this.editorStore.graphManagerState.graph,
+            )) as TestDataGenerationResult;
+          if (testDataGenerationResult.data[0]) {
+            service_setConnectionTestDataEmbeddedData(
+              this.connectionData,
+              testDataGenerationResult.data[0],
+              this.editorStore.changeDetectionState.observerContext,
+            );
+            this.embeddedEditorState = new EmbeddedDataEditorState(
+              this.testDataState.editorStore,
+              this.connectionData.testData,
+            );
+          } else {
+            this.editorStore.applicationStore.notificationService.notifyError(
```

Comment:
lets fix the error messaging here.
we should throw in the above for the reason we can't generate. 
so we should do something like
AsserTrue(testDataGenerationResult.length === 1, 'Expected generated data to be of size 1');
also if the runtime is not a pointer, same thing. 

The catch below will be responsible to catch and notify user of the error. you dont need the notify up here in the try logic.